File: src/main/java/com/hazza/LeetCodeByJava/SegmentThree/No111MinimumDepthOfBinaryTree.java
package com.hazza.LeetCodeByJava.SegmentThree;

import com.hazza.LeetCodeByJava.datastructure.TreeNode;

/**
 * Created with IntelliJ IDEA.
 * Description: https://leetcode.com/problems/minimum-depth-of-binary-tree/description/
 * User: HazzaCheng
 * Contact: hazzacheng@gmail.com
 * Date: 18-1-23
 * Time: 2:11 PM
 */
public class No111MinimumDepthOfBinaryTree {
    public int minDepth(TreeNode root) {
        if(root == null) return 0;
        int leftDepth = minDepth(root.left);
        int rightDepth = minDepth(root.right);
        return (leftDepth == 0 || rightDepth == 0) ? leftDepth + rightDepth + 1: Math.min(leftDepth,rightDepth) + 1;
    }

}

File: src/main/python/leetcode_by_python/tree/No111_Minimum_Depth_of_Binary_Tree.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019-10-19
from src.main.python.leetcode_by_python.data_structure import TreeNode


class Solution:
    def minDepth(self, root: TreeNode) -> int:
        if not root:
            return 0
        depths = list(map(self.minDepth, (root.left, root.right)))

        return 1 + (min(depths) or max(depths))

